pennylane/devices/default_qubit.py

```diff
@@ -253,7 +263,8 @@ def _apply_operation(self, state, operation):
         wires = operation.wires
 
         if operation.base_name in self._apply_ops:
-            axes = self.wires.indices(wires)
+            shift = int(self._ndim(state) > self.num_wires)
+            axes = [ax + shift for ax in self.wires.indices(wires)]
```

Comment:
I am a little confused here. I don't see how adding more wires would increase the `ndim(state)`? 

eg. if `num_wires = 1`, I would imagine the state to be something like `state = array([1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j])` and thus `ndim(state) = 1`. Then if `num_wires=2` I imagine `state = array([1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j])` which has `ndim(state) = 1`.

Reply:
Intuitively you are right. However, the device stores its state as a `(2, 2, 2, 2...)` (with `num_wires` many `2`s) tensor, instead of a `(2**num_wires,)` vector (see for example `_create_basis_state`).
Therefore, without broadcasting, there are `num_wires` axes in `self._state`, and with broadcasting, there are `num_wires+1` :)

Reply:
The reason for this is performance, as far as I can tell: The high-dimensional tensor notation allows for applying, say, a single-qubit gate matrix to the state, without blowing it up to the full `2**num_wires`-dimensional matrix it would correspond to on the whole state space.